scripting/include/srccoop/classdef.inc

```diff
@@ -742,6 +820,108 @@ methodmap CBaseEntity
 	{
 		SetEntProp(this.GetEntIndex(), Prop_Data, "m_iMaxHealth", iMaxHealth);
 	}
+	public bool FVisibleEntity(CBaseEntity pEntity, int iTraceMask = MASK_BLOCKLOS, CBaseEntity& pBlocker = NULL_CBASEENTITY)
+	{
+		if (pEntity.GetFlags() & FL_NOTARGET)
+			return false;
+
+		float vec3LookerOrigin[3];
+		this.GetEyePosition(vec3LookerOrigin);
+		float vec3TargetOrigin[3];
+		pEntity.GetEyePosition(vec3TargetOrigin);
+
+		CTraceRay pTraceRay;
+		if (FindConVar("ai_los_mode").BoolValue)	// TODO: Cache this ConVar.
+		{
+			pTraceRay = new CTraceRay(vec3LookerOrigin, vec3TargetOrigin, iTraceMask, RayType_EndPoint, TraceEntityFilter_FVisibleAiLosEnabled, this.GetEntIndex());
+		}
+		else
+		{
+			// If we're doing an LOS search, include NPCs.
+			if (iTraceMask == MASK_BLOCKLOS)
+			{
+				iTraceMask = MASK_BLOCKLOS_AND_NPCS;
+			}
+
+			// Player sees through nodraw
+			if (this.IsClassPlayer())
+			{
+				iTraceMask &= ~CONTENTS_BLOCKLOS;
+			}
+
+			pTraceRay = new CTraceRay(vec3LookerOrigin, vec3TargetOrigin, iTraceMask, RayType_EndPoint, TraceEntityFilter_FVisible, this.GetEntIndex());
+		}
+
+		if (pTraceRay.GetFraction() != 1.0 || pTraceRay.StartedSolid())
+		{
+			CBaseEntity pTracedEntity = pTraceRay.GetEntity();
+
+			// If we hit the entity we're looking for, it's visible
+			if (pEntity == pTracedEntity)
+			{
+				pTraceRay.Close();
+				return true;
+			}
+
+			if (pEntity.IsClassPlayer())
+			{
+				CBasePlayer pPlayer = view_as<CBasePlayer>(pEntity);
+				if (pPlayer.GetVehicle() == pTracedEntity)
+				{
+					// Got line of sight on the vehicle the player is driving!
+					pTraceRay.Close();
+					return true;
+				}
+			}
+
+			// Line of sight is not established
+			pBlocker = pTracedEntity;
+		}
+
+		pTraceRay.Close();
+		return false;
+	}
+	public bool FVisiblePosition(const float vec3TargetOrigin[3], int iTraceMask = MASK_BLOCKLOS, CBaseEntity& pBlocker = NULL_CBASEENTITY)
+	{
+		float vec3LookerOrigin[3];
+		this.GetEyePosition(vec3LookerOrigin);
+
+		CTraceRay pTraceRay;
+		if (FindConVar("ai_los_mode").BoolValue)	// TODO: Cache this ConVar.
```

Comment:
can be static

Reply:
Madethe  `FindConVar` a global variable that gets initialized once. Fixed.